src/Data/Array/Mutable.idr

```diff
@@ -200,6 +200,24 @@ parameters {n : Nat}
         _   # t := copy r 0 0 n @{reflexive} @{lteAddLeft n} tgt t
      in tgt # t
 
+--------------------------------------------------------------------------------
+--          Appending Arrays
+--------------------------------------------------------------------------------
+
+parameters {m, n : Nat}
+           (p : MArray s m a)
+           (q : MArray s n a)
+
+  ||| Allocates a new mutable array and adds the elements from `p`
+  ||| at its beginning, followed by adding the elements from `q`.
+  export
+  mappend : (deflt : a) -> F1 s (MArray s (n+m) a)
+  mappend deflt t =
+    let tgt # t := marray1 (n+m) deflt t
+        _   # t := copy p 0 0 m @{reflexive} @{lteAddLeft m} tgt t
+        _   # t := copy q 0 m n @{lteAddLeft (m+n)} tgt t
+      in tgt # t
+
```

Comment:
There's no need for a default argument here. I suggest do use `unsafeMArray1` to allocate the new array and drop the `deflt` argument.